find_max.rb

```diff
@@ -0,0 +1,19 @@
+
+# Find the maximum element between index 0 and max_index, exclusive
+
+def find_max(array, max_index)
+  max = nil
+  if max_index == 0
+    puts "Array is empty"
+  else
+    max = array[0]
+    index = 0
+    while index < max_index
+      if max < array[index]
```

Comment:
since you're increasing index until index < max_index is false, you should check that the array has at least max_index number of indices